src/pages/SignUp/Field.tsx

```diff
@@ -0,0 +1,49 @@
+import Icon, { IconType } from "components/Icon";
+import { unpack } from "components/form/helpers";
+import { Classes } from "components/form/types";
+import { FieldValues, Path, get, useFormContext } from "react-hook-form";
+import { fieldClasses } from "./constants";
+
+type Props<T extends FieldValues> = {
+  classes?: Classes;
+  name: Path<T>;
+  placeholder: string;
+  icon?: IconType;
+};
+
+export default function Field<T extends FieldValues>(props: Props<T>) {
+  const {
+    register,
+    formState: { errors },
+  } = useFormContext<T>();
+
+  const errorMsg = get(errors, props.name)?.message;
+
+  const { container, input, error } = unpack(props.classes);
+
+  return (
+    <div className={container}>
+      <div
+        className={`grid ${
+          props.icon ? "grid-cols-[auto_1fr]" : ""
+        } ${fieldClasses}`}
+      >
+        {props.icon && (
+          <Icon type={props.icon} className="ml-5 text-navy-l3" size={20} />
+        )}
+        <input
+          {...register(props.name)}
+          type="text"
+          className={`w-full h-full placeholder:font-medium placeholder:font-heading placeholder:text-navy-l3 focus:outline-none bg-transparent ${
+            props.icon ? "pr-5" : "px-5"
+          } ${input}`}
+          placeholder={props.placeholder}
+          aria-invalid={!!errorMsg}
+        />
+      </div>
+      {errorMsg && (
+        <p className={`text-xs text-[#C52828] mt-1.5 ${error}`}>{errorMsg}</p>
```

Comment:
@SovereignAndrey, looked at the colors, and I think we can use the standard red they provided in place of this stray `#C52828`
<img width="802" alt="image" src="https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/29f150a1-0c51-4e4f-8040-143b28fe429b">

they have very small difference in `HSL` values so I don't think we need two reds for the same purpose
<img width="252" alt="image" src="https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/91498891-270d-4a6f-b9c5-6a5c878577e0">

Reply:
@ap-justin you're suggesting we use `red-d3` instead, right?